detection_rules/rule.py

```diff
@@ -1159,11 +1159,14 @@ def flattened_dict(self) -> dict:
         flattened.update(self.metadata.to_dict())
         return flattened
 
-    def to_api_format(self, include_version=True) -> dict:
+    def to_api_format(self, include_version: bool = True, include_metadata: bool = False) -> dict:
         """Convert the TOML rule to the API format."""
         converted_data = self.to_dict()['rule']
         converted = self._post_dict_conversion(converted_data)
 
+        if include_metadata:
+            converted["meta"] = self.to_dict()['metadata']
```

Comment:
We call to_dict a couple lines above where we interact with the metadata. Can we just reuse that? For example.

```suggestion
        rule_dict = self.to_dict()
        converted_data = rule_dict['rule']
        converted = self._post_dict_conversion(converted_data)

        if include_metadata:
            converted["meta"] = rule_dict['metadata']
```

Reply:
Yes we can.